qiskit/validation/result.py

```diff
@@ -0,0 +1,133 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2018, IBM.
+#
+# This source code is licensed under the Apache License, Version 2.0 found in
+# the LICENSE.txt file in the root directory of this source tree.
+
+"""Schema and Model for schema-conformant Results.
+
+TODO: Note this file is temporary, and will be removed before 0.7, replacing
+      qiskit.qobj.result and qiskit.result.
+"""
+
+from marshmallow.fields import Boolean, DateTime, Integer, List, Nested, Raw, String
+from marshmallow.validate import Length, OneOf, Regexp, Range
+
+from qiskit.validation.base import BaseModel, BaseSchema, ObjSchema, bind_schema
+from qiskit.validation.fields import Complex, ByType
+from qiskit.validation.validate import PatternProperties
+
+
+class ExperimentResultDataSchema(BaseSchema):
+    """Schema for ExperimentResultData."""
+
+    counts = Nested(ObjSchema,
+                    validate=PatternProperties(
+                        {Regexp('^0x([0-9A-Fa-f])+$'): Integer()}))
+    snapshots = Nested(ObjSchema)
+    memory = List(Raw(),
+                  validate=Length(min=1))
+    statevector = List(List(Complex(),
+                            validate=Length(min=1)))
+    unitary = List(List(List(Complex(),
```

Comment:
Given the level of nesting of the fields, I think we could increase legibility by intending field definitions as:
```python
unitary = List(
  List(
    List(Complex, validate=Length(min=1)),
    validate=Length(min=1)
  ),
  validate=Length(min=1)
)
```